src/Factories/DocumentFactory.php

```diff
@@ -32,7 +33,7 @@ class_basename($model)
                 ));
             }
 
-            return new Document($documentId, $documentContent);
+            return new Document($documentId, $documentContent, $documentRouting);
```

Comment:
As mentioned in other PR, I'd prefer to have the routing path as `DocumentManager::index()`/`DocumentManager::delete()` methods parameter:

```php
class Engine
{
    public function update($models) 
    {
         // ..
        $index = $models->first()->searchableAs();
        $documents = $this->documentFactory->makeFromModels($models);
        $routingPath = in_array(ShardRouting::class, class_uses_recursive($model)) ? $model->getRoutingPath() : null;

        $this->documentManager->index($index, $documents->all(), $this->refreshDocuments, $routingPath);
    }
}
```

Reply:
Even though I mentioned, that `ShardRouting` must be in Plus package, the `Engine` changes can be done here. We just shouldn't expose this functionality directly in this package, because its main idea is to be fully backward compatible with other Scout drivers.

Reply:
@babenkoivan I see your point, thx for the suggestion.

I would like to expand on it though.
If we want elastic-scout-driver to be fully compatible with other scout drivers wouldn't it be preferable to reimplement/override the engine in elastic-driver-plus? Imho that would simplify the development of both libraries since required changed in plus wouldn't need to leak into elastic-scout-driver.

We could have elastic-scout-driver be the scout compatible lib. The elastic-scout-driver-plus would depend on elastic-scout-driver and override/extend things as it sees fit. So a developer that needs all the elastic goodies would only require elastic-scout-driver-plus and not both libraries.

Reply:
@spiritinlife you are right, ideally we override this method in `ElasticScoutDriverPlus\Decorators\EngineDecorator`.

Reply:
@spiritinlife we also need to override `Engine::flush()` method, it uses `deleteByQuery`, which doesn't support custom routing.